cmd/kubeadm/app/phases/upgrade/compute_test.go

```diff
@@ -826,3 +826,56 @@ func TestKubeletUpgrade(t *testing.T) {
 		}
 	}
 }
+
+func TestGetBranchFromVersion(t *testing.T) {
+	testCases := []struct {
+		version         string
+		expectedVersion string
+	}{
+		{
+			version:         "v1.9.5",
+			expectedVersion: "1.9",
+		},
+		{
+			version:         "v1.9.0-alpha.2",
+			expectedVersion: "1.9",
+		},
+		{
+			version:         "v1.9.0-beta.0",
+			expectedVersion: "1.9",
+		},
+		{
+			version:         "v1.9.0-rc.1",
+			expectedVersion: "1.9",
+		},
+		{
+			version:         "v1.12.5",
+			expectedVersion: "1.12",
+		},
+		{
+			version:         "v1.11.0-alpha.0",
+			expectedVersion: "1.11",
+		},
+
+		{
+			version:         "v1.11.0-beta.1",
+			expectedVersion: "1.11",
+		},
+		{
+			version:         "v1.11.0-rc.0",
+			expectedVersion: "1.11",
+		},
+		{
+			version:         "1.12.5",
+			expectedVersion: "1.12",
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
subtests will be better!

Reply:
@wgliang We have a lot of tests that look like this, so this meets the current standard.
But yeah for sure subtests would probably be better, please do contribute that to our tests to higher the quality :wink: